src/phylum/ci/ci_base.py

```diff
@@ -181,6 +184,30 @@ def depfiles(self) -> Depfiles:
             self.returncode = ReturnCode.NO_DEPFILES_PROVIDED
         raise SystemExit(self.returncode)
 
+    def _exclude_depfiles(self, provided_depfiles: DepfileEntries) -> DepfileEntries:
+        """Apply exclusion patterns to provided dependency files and return the remaining ones."""
+        arg_exclusions: Optional[list[list[str]]] = self.args.exclude
+        if not arg_exclusions:
+            LOG.debug("No dependency file exclusion patterns provided.")
+            return provided_depfiles
+
+        # flatten the list of lists
+        provided_arg_exclusions = list(chain.from_iterable(arg_exclusions))
+        LOG.debug("Exclusion patterns provided as arguments: %s", provided_arg_exclusions)
+
+        excluded_depfiles: DepfileEntries = []
+        for depfile in provided_depfiles:
+            for exclusion in provided_arg_exclusions:
+                if depfile.path.match(exclusion):
```

Comment:
It seems like we're still missing the ability to completely exclude a given directory...

For example, if I have a `test` folder and I want to exclude it, I can write `test/*`, but that won't exclude `/test/a/package-lock.json`. So I also need `test/*/*`, `test/*/*/*`, and so on.

I wonder if we should be matching `depfile` and all parents of `depfile` for a match to allow simply writing the pattern as `test` to ignore everything in or under a folder named `test`

Reply:
> I wonder if we should be matching `depfile` and all parents of `depfile` for a match to allow simply writing the pattern as `test` to ignore everything in or under a folder named `test`

Implementing this would mean deviating from the "glob-style" format, which might introduce subtle bugs, make documenting usage more complicated, and/or make it slightly more confusing to understand as a user. It is possible but not desired.

Instead, alternatives are suggested for proceeding:

* Keep this feature as it is and rely on users to supply the correct patterns
* Switch away from glob-style matching to gitignore-style matching
  * This can be done with the [`pathspec` third-party library](https://python-path-specification.readthedocs.io/en/latest/readme.html)

@kylewillmon, is one of these alternatives acceptable or would you prefer something else?

Reply:
> Keep this feature as it is and rely on users to supply the correct patterns

I'm not sure if this is a reasonable option... There is no "correct pattern" for ignoring an entire directory with glob-style patterns. So by choosing this option, we are precluding that seemingly desirable feature.

I usually see glob-style for "include" patterns while gitignore-style is typically used in "exclude" patterns. So that also makes glob-style feel like an odd choice here.

Finally, the CLI ignores files based on the content `.gitignore` and `.ignore` files. And those files obviously use a gitignore-style. So that would make sense as a matter of uniformity across the product.

I understand that we want this feature to be done quickly, so I am hesitant to insist on a fundamental design change this late... But if we can afford the time, I think this would be better implemented using gitignore-style.

Reply:
I have come to agree with this sentiment and the change to gitignore-style will be made now. The hope is that it won't take more than a day. I didn't start with gitignore-style matching because it requires a third party library, takes a bit longer to implement and test, and there was a (wrong) initial assumption that the `pathlib` stdlib would allow for `**` recursive wildcard matching.

Reply:
The changes have been made.